File: Two_Pointers/003.Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) 
    {
        vector<int>Map(256,0);
        int i=0;
        int j=0;
        int result=0;
        
        while (j<s.size())
        {
            if (Map[s[j]]==0)
            {
                Map[s[j]]=1;
                j++;
            }
            else if (Map[s[j]]==1)
            {
                Map[s[i]]=0;
                i++;
            }
            
            result = max(result,j-i);
        }
        
        return result;
        
    }
};

File: Hash/3-Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        ventana = set()
        max_longitud = 0
        inicio = 0

        for fin in range(len(s)):
            while s[fin] in ventana:
                ventana.remove(s[inicio])
                inicio += 1
            
            ventana.add(s[fin])
            max_longitud = max(max_longitud, fin - inicio + 1)
        
        return max_longitud
